ee/tabby-webserver/src/service/mod.rs

```diff
@@ -51,15 +52,20 @@ struct ServerContext {
 impl ServerContext {
     pub async fn new(logger: Arc<dyn RawEventLogger>, code: Arc<dyn CodeSearch>) -> Self {
         let db_conn = DbConn::new().await.unwrap();
+        let mail = Arc::new(
+            new_email_service(db_conn.clone())
+                .await
+                .expect("failed to initialize mail service"),
+        );
         Self {
             client: Client::default(),
             completion: worker::WorkerGroup::default(),
             chat: worker::WorkerGroup::default(),
-            mail_service: Arc::new(
-                new_email_service(db_conn.clone())
-                    .await
-                    .expect("failed to initialize mail service"),
-            ),
+            mail_service: mail.clone(),
+            auth_service: Arc::new(AuthenticationServiceImpl {
```

Comment:
instead of making AuthenticationServiceImpl a public struct and degrade the encapsulation, a better way would be extract a function just like `new_email_service` for `new_authentciation_service`, and takes `db` and `email` as the argument.

It's a common pattern to avoid the leaky of implementation detail.